### PR TITLE
[DOC] Updated decoding.SSD documentation and internal variable naming

### PR DESCRIPTION
**Pull request by tsbinns**
Implementation of comments/suggestions in merged PR #11458.

- Fixed incorrect default values in the docstring.
- Reworded the docstring to better reflect the behaviour of the class.
- Changed the naming of internal variables for consistency and clarity.
- Changed a logging message to more accurately reflect the behaviour of the code.

Cheers, Thomas!

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by larsoner** [comment]
@vpeterson can you look? Then we'll mark for merge when you're happy

File: mne/decoding/ssd.py [hunk 1]
```diff
@@ -39,35 +39,36 @@ class SSD(BaseEstimator, TransformerMixin):
...
-        (0 <= shrinkage <= 1). For str options, reg will be passed to
-        method to :func:`mne.compute_covariance`.
+        If not None (same as 'empirical'), allow regularization for covariance
+        estimation. If float, shrinkage is used (0 <= shrinkage <= 1). For str
+        options, reg will be passed to method :func:`mne.compute_covariance`.
     n_components : int | None (default None)
         The number of components to extract from the signal.
-        If n_components is None, no dimensionality reduction is applied.
+        If None, the number of components equal to the rank of the data are
+        returned (see :attr:`rank`).
```
Comment (larsoner) [inline mne/decoding/ssd.py:48]: Sphinx has trouble resolving this so let's just use code mode for now
```suggestion
        returned (see ``rank``).
```

File: mne/decoding/ssd.py [hunk 2]
```diff
@@ -39,35 +39,36 @@ class SSD(BaseEstimator, TransformerMixin):
...
-       to the spectral ratio.
-       See Eq. (24) in :footcite:`NikulinEtAl2011`.
-    return_filtered : bool (default True)
-        If return_filtered is True, data is bandpassed and projected onto
-        the SSD components.
+    sort_by_spectral_ratio : bool (default True)
+        If set to True, the components are sorted according to the spectral
+        ratio.
+        See Eq. (24) in :footcite:`NikulinEtAl2011`.
+    return_filtered : bool (default False)
```
Comment (vpeterson) [inline mne/decoding/ssd.py:55]: reasons for changing these defaults?!

File: mne/decoding/ssd.py [hunk 3]
```diff
@@ -39,35 +39,36 @@ class SSD(BaseEstimator, TransformerMixin):
...
        See :func:`mne.time_frequency.psd_array_welch` for more information.
     cov_method_params : dict | None (default None)
         As in :class:`mne.decoding.SPoC`
         The default is None.
     rank : None | dict | ‘info’ | ‘full’
         As in :class:`mne.decoding.SPoC`
         This controls the rank computation that can be read from the
-        measurement info or estimated from the data.
+        measurement info or estimated from the data, which determines the
+        number of available components.
```
Comment (vpeterson) [inline mne/decoding/ssd.py:71]: I would say "which determines the maximum possible number of components"

File: mne/decoding/ssd.py (hunk 2 above)
Comment (tsbinns) [reply mne/decoding/ssd.py:55]: In `__init__`, `sort_by_spectral_ratio` is default `True`, and `return_filtered` is default `False`:
https://github.com/mne-tools/mne-python/blob/4ee23004264b63ba2861ca70a4c777aee141de97/mne/decoding/ssd.py#L86-L89

however the existing documentation lists this the other way around:
https://github.com/mne-tools/mne-python/blob/4ee23004264b63ba2861ca70a4c777aee141de97/mne/decoding/ssd.py#L51-L57

so this change was just to have the documentation reflect the actual default values.

**Pull request merged**